packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js

```diff
@@ -2248,46 +2248,155 @@ QUnit.module('aria accessibility', () => {
+            $input = $dropDownEditor.find(`.${TEXT_EDITOR_INPUT_CLASS}`);
+            assert.strictEqual($input.attr(attribute), value, `${attribute} attribute should remain ${value} after typing`);
 
-        keyboardMock($input)
-            .type('a');
+            keyboardMock($input)
+                .caret(1)
+                .press('backspace');
 
-        $input = $dropDownEditor.find(`.${TEXT_EDITOR_INPUT_CLASS}`);
+            $input = $dropDownEditor.find(`.${TEXT_EDITOR_INPUT_CLASS}`);
+            assert.strictEqual($input.attr(attribute), value, `${attribute} attribute should remain ${value} after deleting`);
+        });
+    });
+
+    QUnit.module('aria-invalid', {}, () => {
+        [
+            { valueRequired: true, emptyValue: 'true', nonEmptyValue: undefined },
+            { valueRequired: false, emptyValue: undefined, nonEmptyValue: undefined }
+        ].forEach(({ valueRequired, emptyValue, nonEmptyValue }) => {
+            QUnit.test(`component with fieldTemplate should have proper aria-invalid attribute when validator is used and value is ${!valueRequired ? 'not' : ''} required (T1230706)`, function(assert) {
+                const clock = sinon.useFakeTimers();
+
+                const $dropDownEditor = $('#dropDownEditorSecond').dxDropDownEditor({
+                    dataSource: ['one', 'two', 'three'],
+                    searchEnabled: true,
+                    fieldTemplate: 'field',
+                    templatesRenderAsynchronously: true,
+                    integrationOptions: {
+                        templates: {
+                            field: {
+                                render: function({ model, container, onRendered }) {
+                                    const $input = $('<div>').appendTo(container);
+
+                                    setTimeout(() => {
+                                        $input.dxTextBox({ value: model });
+                                        onRendered();
+                                    }, 0);
+                                }
+                            }
+                        }
+                    },
+                    valueChangeEvent: 'keyup',
+                }).dxValidator({
+                    validationRules: valueRequired ? [{ type: 'required', message: 'required' }] : [],
+                });
 
-        assert.strictEqual($input.attr('aria-required'), 'true', 'aria-required attribute should remain true after typing');
+                clock.tick(500);
 
-        assert.strictEqual($input.attr('aria-haspopup'), 'true', 'aria-haspopup attribute should retain to true after typing');
+                let $input = $dropDownEditor.find(`.${TEXT_EDITOR_INPUT_CLASS}`);
 
-        assert.strictEqual($input.attr('aria-autocomplete'), 'none', 'aria-autocomplete attribute should retain to none after typing');
+                assert.equal($input.val(), '', 'input value is empty');
```

Comment:
test looks too big, let's at least remove excess asserts with input value check on init, there's no reason to check it in these tests

Reply:
[64e8b3b](https://github.com/DevExpress/DevExtreme/pull/27402/commits/64e8b3b5c34f22d2164afbded0e447dc68d5eef2)